File: src/main/java/burrow/furniture/hoard/command/NewCommand.java
package burrow.furniture.hoard.command;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.furniture.hoard.HoardFurniture;
import org.springframework.lang.NonNull;
import picocli.CommandLine;

import java.util.HashMap;

@CommandLine.Command(name = "new", description = "Create a new entry.")
@CommandType(HoardFurniture.COMMAND_TYPE)
public class NewCommand extends Command {
    @CommandLine.Parameters(arity = "0..*")
    private String[] params;

    public NewCommand(@NonNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() {
        if (params != null && params.length % 2 == 1) {
            buffer.append("Invalid number of arguments: ").append(params.length);
            return CommandLine.ExitCode.USAGE;
        }

        final var properties = new HashMap<String, String>();
        if (params != null) {
            for (int i = 0; i < params.length / 2; ++i) {
                properties.put(params[i * 2], params[i * 2 + 1]);
            }
        }

        final var hoardFurniture = use(HoardFurniture.class);
        final var entry = hoardFurniture.getHoard().create(properties);
        buffer.append(hoardFurniture.entryToString(entry, commandContext));

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/furnishing/hoard/command/NewCommand.kt
package burrow.furnishing.hoard.command

import burrow.furnishing.hoard.Hoard
import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine

@CommandLine.Command(
    name = "new",
    description = ["Creates a new entry."],
)
class NewCommand(data: CommandData) : Command(data) {
    @CommandLine.Parameters(arity = "0..*")
    private var params: Array<String> = emptyArray()

    override fun call(): Int {
        if (params.isEmpty() && params.size % 2 == 1) {
            stderr.println("Invalid number of arguments: ${params.size}")
            return CommandLine.ExitCode.USAGE
        }

        val properties = mutableMapOf<String, String>()
        for (i in 0 until params.size / 2) {
            properties[params[i * 2]] = params[i * 2 + 1]
        }

        val entry = use(Hoard::class).create(properties)
        stdout.println(entry.id)

        return CommandLine.ExitCode.OK;
    }
}